src/main/java/org/jabref/model/entry/LinkedFile.java

```diff
@@ -19,42 +27,49 @@
 public class LinkedFile implements Serializable {
 
     private static final LinkedFile NULL_OBJECT = new LinkedFile("", "", "");
-    private String description;
-    private String link;
-    private String fileType;
+    private final StringProperty description = new SimpleStringProperty();
+    private final StringProperty link = new SimpleStringProperty();
+    private final StringProperty fileType = new SimpleStringProperty();
+    private final DoubleProperty downloadProgress = new SimpleDoubleProperty(-1);
+    private final BooleanProperty isAutomaticallyFound = new SimpleBooleanProperty(false);
 
     public LinkedFile(String description, String link, String fileType) {
-        this.description = Objects.requireNonNull(description);
-        this.link = Objects.requireNonNull(link);
-        this.fileType = Objects.requireNonNull(fileType);
+        this.description.setValue(Objects.requireNonNull(description));
+        this.link.setValue(Objects.requireNonNull(link));
+        this.fileType.setValue(Objects.requireNonNull(fileType));
     }
 
     public LinkedFile(String description, URL link, String fileType) {
         this(description, Objects.requireNonNull(link).toString(), fileType);
     }
 
     public String getFileType() {
-        return fileType;
+        return fileType.get();
     }
 
     public void setFileType(String fileType) {
-        this.fileType = fileType;
+        this.fileType.setValue(fileType);
     }
 
     public String getDescription() {
-        return description;
+        return description.get();
     }
 
     public void setDescription(String description) {
-        this.description = description;
+        this.description.setValue(description);
+
     }
 
     public String getLink() {
-        return link;
+        return link.get();
     }
 
     public void setLink(String link) {
-        this.link = link;
+        this.link.setValue(link);
+    }
+
+    public Observable[] getObservables() {
+        return new Observable[] {this.downloadProgress, this.isAutomaticallyFound};
```

Comment:
you need to return `description, link, fileType` here (and move `downloadProgress` and `isAutomaticallyFound` back to the viewmodel).

The observable tell javafx which properties mark the object as changed, i.e. if `description` is changed this should mark the `LinkedFile` as changed and thus also `LinkedFileViewModel` and thus finally the list https://github.com/JabRef/jabref/blob/23c6ac6308dc4df72057eec0806570f635c1ed0b/src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditorViewModel.java#L55.
For more details see https://docs.oracle.com/javase/8/javafx/api/javafx/collections/FXCollections.html#observableList-java.util.List-javafx.util.Callback- or https://stackoverflow.com/questions/31687642/callback-and-extractors-for-javafx-observablelist